src/CSStrikes.sol

```diff
@@ -6,23 +6,27 @@ pragma solidity 0.8.24;
 import { MerkleProof } from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
 
 import { ICSModule } from "./interfaces/ICSModule.sol";
+import { ICSEjector } from "./interfaces/ICSEjector.sol";
 import { ICSStrikes } from "./interfaces/ICSStrikes.sol";
 
 /// @author vgorkavenko
 contract CSStrikes is ICSStrikes {
     address public immutable ORACLE;
     ICSModule public immutable MODULE;
+    ICSEjector public immutable EJECTOR;
 
     /// @notice The latest Merkle Tree root
     bytes32 public treeRoot;
 
     /// @notice CID of the last published Merkle tree
     string public treeCid;
 
-    constructor(address module, address oracle) {
+    constructor(address module, address ejector, address oracle) {
         if (module == address(0)) revert ZeroModuleAddress();
+        if (ejector == address(0)) revert ZeroEjectorAddress();
         if (oracle == address(0)) revert ZeroOracleAddress();
         MODULE = ICSModule(module);
```

Comment:
`MODULE` is no longer needed here. We can call `EJECTOR.MODULE().getSigningKeys(...)`

Reply:
We can. But why not have it explicitly?

Reply:
lets just fetch it from the ejector contract to reduce checks and inputs